api/src/main/java/org/openmrs/module/eptsreports/reporting/library/dimensions/TxNewAgeDimensionCohortStrategy.java

```diff
@@ -0,0 +1,47 @@
+package org.openmrs.module.eptsreports.reporting.library.dimensions;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import org.openmrs.Location;
+import org.openmrs.module.eptsreports.reporting.library.cohorts.AgeCohortQueries;
+import org.openmrs.module.eptsreports.reporting.library.cohorts.TxNewCohortQueries;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Mapped;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * Uses age cohorts that determine age on ART start date.
+ */
+@Component("txnew")
+public class TxNewAgeDimensionCohortStrategy implements AgeDimensionCohortStrategy {
+
+  @Autowired private TxNewCohortQueries txNewCohortQueries;
+
+  @Autowired private AgeCohortQueries ageCohortQueries;
+
+  @Override
+  public Mapped<CohortDefinition> createXtoYAgeCohort(String name, Integer minAge, Integer maxAge) {
+    CohortDefinition ageCohort =
+        txNewCohortQueries.createXtoYAgeOnArtStartDateCohort(name, minAge, maxAge);
+    return EptsReportUtils.map(
+        ageCohort, "location=${location},onOrAfter=${startDate},onOrBefore=${endDate}");
+  }
+
+  @Override
+  public Mapped<CohortDefinition> createUnknownAgeCohort() {
```

Comment:
There is an `getUnknownAgeCohort` cohort function in genericCohortQueries.java. Please remove that and update all uses of it to this one from ageCohortQueries